hazelcast/src/test/java/com/hazelcast/cache/CacheTestUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache;
+
+import com.hazelcast.cache.impl.CacheService;
+import com.hazelcast.cache.impl.HazelcastServerCacheManager;
+import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
+import com.hazelcast.cache.impl.ICacheRecordStore;
+import com.hazelcast.cache.impl.record.CacheRecord;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.partition.InternalPartitionService;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.serialization.SerializationService;
+
+import javax.cache.spi.CachingProvider;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.hazelcast.test.HazelcastTestSupport.getNodeEngineImpl;
+
+public final class CacheTestUtil {
+
+    private CacheTestUtil() {
+    }
+
+    /**
+     * Returns the backup entries of an {@link ICache} by a given cache name.
+     *
+     * @param instances the {@link HazelcastInstance} array to gather the data from
+     * @param cacheName   the cache name
+     * @param <K>       type of the key
+     * @param <V>       type of the value
+     * @return a {@link Map} with the backup entries
+     */
+    public static <K, V> Map<K, V> getBackupCache(HazelcastInstance[] instances, String cacheName) {
```

Comment:
can't get the exact goal here? What if icache has more than 1 backup? Maybe all backups are not sync yet?

Reply:
Basically all data structures were missing thorough backup tests, especially the content of the backups. In some places the number of backups was checked, but never if the content was correct. This is essential to check if the split-brain healing is working correctly.

For the non-partitioned data structures the logic to retrieve the backups is a bit simpler (and there I actually give the replica index to the method, to retrieve the exact backup replica). For the partitioned data structures I'm happy enough to retrieve all backup replicas. The tests are configured with a single sync backup. This is good enough to check if the content of the backups is what is expected. This helped me to correctly implement the backup operations in all split-brain healing implementations (and it almost never worked on the first try).